test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/SingleFileVerifier.java

```diff
@@ -99,6 +99,11 @@ static SingleFileVerifier create(Path sourceFilePath, Charset encoding) {
    */
   void assertNoIssues();
 
+  /**
+   * Run the comparison and expect to find no issue. Allows Noncompliant comments to be present in the file.
```

Comment:
Let's also update the comment of `assertNoIssues()` to explicitly mention that noncompliant comments are not allowed.